src/settings.cc

```diff
@@ -128,19 +135,19 @@ SettingsEntry* Settings::getSettingEntryByName(const std::string &name)
 	return nullptr;
 }
 
-const Value &Settings::defaultValue(const SettingsEntry& entry) const
+Value Settings::defaultValue(const SettingsEntry& entry) const
 {
-	return entry._default;
+	return entry._default.clone();
 }
 
-const Value &Settings::get(const SettingsEntry& entry) const
+Value Settings::get(const SettingsEntry& entry) const
 {
-	return entry._value;
+	return entry._value.clone();
 }
 
 void Settings::set(SettingsEntry& entry, const Value &val)
 {
-	entry._value = val;
+	entry._value = val.clone();
```

Comment:
In this case, you're _always_ taking a copy of the object. We can automate this by passing the `val` argument by value and `std::move()` it here. This has an added benefit of allowing any object literal to be move-constructed when calling this function, and will reduce the need for having to manually `move()` it at the call site. See other comment for call site